vendor/bat-native-ledger/src/bat/ledger/internal/legacy/client_properties.cc

```diff
@@ -47,4 +68,132 @@ bool ClientProperties::operator!=(
   return !(*this == rhs);
 }
 
+base::Value::Dict ClientProperties::ToValue() const {
+  base::Value::Dict dict;
+  dict.Set(kWalletInfoKey, wallet_info.ToValue());
+  dict.Set(kBootTimestampKey, base::Int64ToValue(boot_timestamp));
+  dict.Set(kReconcileTimestampKey, base::Int64ToValue(reconcile_timestamp));
+  dict.Set(kFeeAmountKey, fee_amount);
+  dict.Set(kUserChangedFeeKey, user_changed_fee);
+  dict.Set(kRewardsEnabledKey, rewards_enabled);
+  dict.Set(kAutoContributeKey, auto_contribute);
+
+  base::Value::Dict inline_tips_dict;
+  for (const auto& [key, value] : inline_tips) {
+    inline_tips_dict.Set(key, value);
+  }
+  dict.Set(kInlineTipsKey, std::move(inline_tips_dict));
+
+  return dict;
+}
+
+bool ClientProperties::FromValue(const base::Value::Dict& dict) {
+  // Wallet Info
+  const auto* wallet_info_dict = dict.FindDict(kWalletInfoKey);
+  if (!wallet_info_dict) {
+    NOTREACHED();
+    return false;
+  }
+  if (!wallet_info.FromValue(*wallet_info_dict)) {
+    NOTREACHED();
+    return false;
+  }
+
+  // Boot Timestamp (There is no support for uint64_t. Writing JSON with such
+  // types violates the spec. As we need a uint64_t, we need to use an unsigned
+  // int and cast to a uint64_t)
+  if (auto value = base::ValueToInt64(dict.Find(kBootTimestampKey))) {
+    boot_timestamp = *value;
+  } else if (auto value = dict.FindDouble(kBootTimestampKey)) {
+    boot_timestamp = static_cast<uint64_t>(*value);
+  } else {
+    NOTREACHED();
+    return false;
+  }
+
+  // Reconcile Timestamp (There is no support for uint64_t. Writing JSON with
+  // such types violates the spec. As we need a uint64_t, we need to use an
+  // unsigned int and cast to a uint64_t)
+  if (auto value = base::ValueToInt64(dict.Find(kReconcileTimestampKey))) {
+    reconcile_timestamp = *value;
+  } else if (auto value = dict.FindDouble(kReconcileTimestampKey)) {
+    reconcile_timestamp = static_cast<uint64_t>(*value);
+  } else {
+    NOTREACHED();
+    return false;
+  }
+
+  // Fee Amount
+  if (auto value = dict.FindDouble(kFeeAmountKey)) {
+    fee_amount = *value;
+  } else {
+    NOTREACHED();
+    return false;
+  }
+
+  // User Changed Fee
+  if (auto value = dict.FindBool(kUserChangedFeeKey)) {
+    user_changed_fee = *value;
+  } else {
+    NOTREACHED();
+    return false;
+  }
+
+  // Auto Contribute
+  if (auto value = dict.FindBool(kAutoContributeKey)) {
+    auto_contribute = *value;
+  } else {
+    NOTREACHED();
+    return false;
+  }
+
+  // Rewards Enabled
+  if (auto value = dict.FindBool(kRewardsEnabledKey)) {
+    rewards_enabled = *value;
+  } else {
+    NOTREACHED();
+    return false;
+  }
+
+  // Inline Tips
+  if (const auto* value = dict.FindDict(kInlineTipsKey)) {
+    for (const auto [key, value] : *value) {
```

Comment:
This makes a copy of both `key` and `value`. I know you can't use `const auto& [key, value]` here, because `base::detail::[const_]dict_iterator` is a proxy iterator, and Clang would warn you that you're extending the lifetime of a `std::pair<const std::string&, [const] base::Value&>` temporary. Just use `auto&& [key, value]`, which passes `-Wrange-loop-bind-reference`.

Reply:
I have already fixed this, but this doesn't make a copy... it merely takes a reference.

Reply:
Sorry, my bad... I think I misunderstood this.

Reply:
I've just tested it in Godbolt and it seems, that yes, it is a reference as I suggested above:
https://godbolt.org/z/sTGzT7jvq
As you can notice k and x are at the same address here. Am I missing something?

Reply:
Nope, that's actually my bad.
So for structured bindings, there's a hidden anonymous variable involved, to which the `auto` type deduction rules and any qualifiers (`const`, `&`, etc.) apply. 
So here:
```
for (const auto [key, value] : *value) { /* ... */ }
```
this is what's happening under the hood:
```
const auto anonymous_variable = rvalue of std::pair<const std::string&, base::Value&> resulting from dereferencing the proxy iterator mentioned above;
aliasname key   = anonymous_variable.first;  // decltype(key)   is the type of anonymous_variable.first  (which is const std::string&)
aliasname value = anonymous_variable.second; // decltype(value) is the type of anonymous_variable.second (which is base::Value&)
```